nifi-nar-bundles/nifi-graph-bundle/nifi-other-graph-services/src/main/java/org/apache/nifi/graph/TinkerpopClientService.java

```diff
@@ -0,0 +1,538 @@
+                for (String clz : classes) {
+                    Class.forName(clz, true, loader);
+                }
+            } catch (Exception ex) {
+                results.add(new ValidationResult.Builder().subject(EXTENSION_CLASSES.getDisplayName()).valid(false).explanation(ex.toString()).build());
+            }
+        }
+
+        boolean standardConfigIsSet = context.getProperty(CONTACT_POINTS).isSet()
+                && context.getProperty(PATH).isSet()
+                && context.getProperty(PORT).isSet();
+        boolean fileIsSet = context.getProperty(REMOTE_OBJECTS_FILE).isSet();
+
+        if (standardConfigIsSet && fileIsSet) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("Cannot set the configuration file and connection details properties at the same time.")
+                    .valid(false)
+                    .build());
+        }
+        if (!standardConfigIsSet && !fileIsSet) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("Connection details properties or the configuration file must be set.").valid(false).build());
+        }
+
+        if (context.getProperty(USER_NAME).isSet() && !context.getProperty(PASSWORD).isSet()) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("When specifying a username, the password must also be set").valid(false).build()
+            );
+        }
+        if (context.getProperty(PASSWORD).isSet() && !context.getProperty(USER_NAME).isSet()) {
+            results.add(new ValidationResult.Builder()
+                    .explanation("When specifying a password, the password must also be set").valid(false).build()
+            );
+        }
+        return results;
+    }
+
+    protected Cluster.Builder setupSSL(ConfigurationContext context, Cluster.Builder builder) {
+        if (context.getProperty(SSL_CONTEXT_SERVICE).isSet()) {
+            SSLContextService service = context.getProperty(SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+            builder
+                    .enableSsl(true)
+                    .keyStore(service.getKeyStoreFile())
+                    .keyStorePassword(service.getKeyStorePassword())
+                    .keyStoreType(service.getKeyStoreType())
+                    .trustStore(service.getTrustStoreFile())
+                    .trustStorePassword(service.getTrustStorePassword());
+            usesSSL = true;
+        }
+
+        return builder;
+    }
+
+
+    public void loadClasses(ConfigurationContext context) {
+        String path = context.getProperty(EXTRA_RESOURCE).getValue();
+        String classList = context.getProperty(EXTENSION_CLASSES).getValue();
+        if (path != null && classList != null && !path.isEmpty() && !classList.isEmpty()) {
+            try {
+                ClassLoader loader = Thread.currentThread().getContextClassLoader();
```

Comment:
See above. Code that worked for me with nifi-mock didn't work in the live framework.